src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -501,21 +492,61 @@ public void SetStateMachine(IAsyncStateMachine stateMachine)
             // cases is we lose the ability to properly step in the debugger, as the debugger uses that
             // object's identity to track this specific builder/state machine.  As such, we proceed to
             // overwrite whatever's there anyway, even if it's non-null.
+#if CORERT
```

Comment:
Maybe worth adding a comment why this is ifdefed out for CoreRT.

BTW: Now that I see it: The ifdefs are not really needed around CreateDebugFinalizableAsyncStateMachineBox. The AOT compilers will drop the unreachable code. The ifdef in this method is the important one, but the other ones can be removed to keep the code simple.

Reply:
I'll add comment. The AOT compiler may see it is unreachable, but it introduces the dependency on `TplEtwProvider.IncompleteAsyncMethod`, which was previously missing. I prefer to #if-out all the code and avoid touching `TplEtwProvider` until a better solution is implemented.

Reply:
ok